Public/Src/Engine/Processes/SandboxedProcessInfo.cs

```diff
@@ -364,5 +366,451 @@ public byte[] GetUnicodeEnvironmentBlock()
+                TimeSpan? timeout = reader.ReadNullableStruct(r => r.ReadTimeSpan());
+                TimeSpan nestedProcessTerminationTimeout = reader.ReadTimeSpan();
+                long pipSemiStableHash = reader.ReadInt64();
+                string timeoutDumpDirectory = reader.ReadNullableString();
+                SandboxKind sandboxKind = (SandboxKind)reader.ReadByte();
+                string pipDescription = reader.ReadNullableString();
+                SandboxedProcessStandardFiles sandboxedProcessStandardFiles = SandboxedProcessStandardFiles.Deserialize(reader);
+                StandardInputInfo standardInputSourceInfo = reader.ReadNullable(r => StandardInputInfo.Deserialize(r));
+                ObserverDescriptor standardObserverDescriptor = reader.ReadNullable(r => ObserverDescriptor.Deserialize(r));
+
+                FileAccessManifest fam = reader.ReadNullable(r => FileAccessManifest.Deserialize(stream));
+
+                return new SandboxedProcessInfo(
+                    new PathTable(),
+                    new StandardFileStorage(sandboxedProcessStandardFiles),
+                    fileName,
+                    fam,
+                    disableConHostSharing,
+                    // TODO: serialize/deserialize container configuration.
+                    containerConfiguration: ContainerConfiguration.DisabledIsolation,
+                    loggingContext: loggingContext,
+                    detoursEventListener: detoursEventListener)
+                {
+                    m_arguments = arguments,
+                    m_commandLine = commandLine,
+                    StandardInputEncoding = standardInputEncoding,
+                    StandardOutputEncoding = standardOutputEncoding,
+                    StandardErrorEncoding = standardErrorEncoding,
+                    WorkingDirectory = workingDirectory,
+                    EnvironmentVariables = buildParameters,
+                    AllowedSurvivingChildProcessNames = allowedSurvivingChildNames,
+                    MaxLengthInMemory = maxLengthInMemory,
+                    Timeout = timeout,
+                    NestedProcessTerminationTimeout = nestedProcessTerminationTimeout,
+                    PipSemiStableHash = pipSemiStableHash,
+                    TimeoutDumpDirectory = timeoutDumpDirectory,
+                    SandboxKind = sandboxKind,
+                    PipDescription = pipDescription,
+                    SandboxedProcessStandardFiles = sandboxedProcessStandardFiles,
+                    StandardInputSourceInfo = standardInputSourceInfo,
+                    StandardObserverDescriptor = standardObserverDescriptor
+                };
+            }
+        }
+
+        #endregion
+
+        #region Extra data structure for serialization
+
+        /// <summary>
+        /// Info about the source of standard input.
+        /// </summary>
+        public class StandardInputInfo : IEquatable<StandardInputInfo>

```

Comment:
>StandardInputInfo [](start = 21, length = 17)

seperate file please

Reply:
I thought they were small at first, but they grew large, and yes, we need to place them in separate files.

---
In reply to: [276819364](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276819364) [](ancestors = 276819364)